mifospay/src/main/java/org/mifos/mobilewallet/mifospay/standinginstruction/ui/SIDetailsActivity.kt

```diff
@@ -0,0 +1,264 @@
+                this.standingInstruction.recurrenceInterval = et_si_edit_interval.text.toString().toInt()
+                val validTillArray = tv_valid_till.text.split("-")
+                this.standingInstruction.validTill = validTillArray.map { it.toInt() }
+
+                // passing the updated standing instruction
+                mStandingInstructionPresenter.updateStandingInstruction(this.standingInstruction)
+            }
+        } else {
+            doSave = true
+            fab.hide()
+            fab.setImageDrawable(resources.getDrawable(R.drawable.ic_save))
+
+            tv_si_amount.visibility = View.GONE
+            til_si_edit_amount.visibility = View.VISIBLE
+
+            tv_edit_pick.visibility = View.VISIBLE
+
+            tv_recurrence_interval.visibility = View.GONE
+            til_si_edit_interval.visibility = View.VISIBLE
+        }
+    }
+
+    /**
+     * Checks the input from the user and returns false if the input is invalid otherwise true
+     */
+    private fun checkInput(): Boolean {
+        if (et_si_edit_amount.text.toString() == "") {
+            showToast(getString(R.string.enter_amount))
+            return false
+        } else if (et_si_edit_amount.text.toString().toDouble() <= 0) {
+            showToast(getString(R.string.enter_valid_amount))
+            return false
+        }
+        if (et_si_edit_interval.text.toString() == "") {
+            showToast(getString(R.string.enter_recurrence_interval))
+            return false
+        } else if (et_si_edit_interval.text.toString().toInt() <= 1) {
+            showToast(getString(R.string.invalid_recurrence_interval))
+            return false
+        }
+        return true
+    }
+
+    override fun showSIDetails(standingInstruction: StandingInstruction) {
+        this.standingInstruction = standingInstruction
+        hideProgressBar()
+        layout_si_details.visibility = View.VISIBLE
+
+        // setting up the layout
+        tv_si_name.text = standingInstruction.name
+        tv_si_id.text = standingInstruction.id.toString()
+        /**
+         * Using hardcoded Currency as response doesn't return the currency
+         */
+        tv_si_amount.text = "${Constants.RUPEE} ${standingInstruction.amount}"
+
+        tv_valid_from.text = "${standingInstruction.validFrom[2]}-" +
+                "${standingInstruction.validFrom[1]}-${standingInstruction.validFrom[0]}"
+        tv_valid_till.text = "${standingInstruction.validTill?.get(2)}-" +
+                "${standingInstruction.validTill?.get(1)}-${standingInstruction.validTill?.get(0)}"
+
+        tv_si_from_name.text = "Name: ${standingInstruction.fromClient.displayName}"
+        tv_si_from_number.text = "Number: ${standingInstruction.fromAccount.accountNo}"
+        tv_si_to_name.text = "Name: ${standingInstruction.toClient.displayName}"
+        tv_si_to_number.text = "Number: ${standingInstruction.toAccount.accountNo}"
```

Comment:
Can we use `formatted string` dynamically in here? Like `<string name="name">Name: %1$s</string>`